modules/siddhi-core/src/test/java/org/wso2/siddhi/core/window/SessionWindowTestCase.java

```diff
@@ -0,0 +1,1065 @@
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test(description = "This test case checks two sessions are processed which  belong to the same session key")
+    public void testSessionWindow12() throws InterruptedException {
+        log.info("SessionWindow Test12: Testing session window, providing events "
+                + "which are in 2 sessions with same session key");
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String purchaseEventStream = ""
+                + "define stream purchaseEventStream (user string, item_number int, price float, quantity int); ";
+
+        String query = ""
+                + "@info(name = 'query0') "
+                + "from purchaseEventStream#window.session(1 sec, user) "
+                + "select * "
+                + "insert all events into outputStream ;";
+
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(purchaseEventStream + query);
+
+        siddhiAppRuntime.addCallback("query0", new QueryCallback() {
+            @Override
+            public void receive(long timestamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timestamp, inEvents, removeEvents);
+                if (inEvents != null) {
+                    inEventCount = inEventCount + inEvents.length;
+                    count.addAndGet(inEvents.length);
+
+                }
+                if (removeEvents != null) {
+                    removeEventCount = removeEventCount + removeEvents.length;
+                    count.addAndGet(removeEvents.length);
+                    if (removeEventCount == 2) {
+                        AssertJUnit.assertTrue("user0".equals(removeEvents[0].getData(0)));
+                        AssertJUnit.assertTrue("user0".equals(removeEvents[1].getData(0)));
+
+                        AssertJUnit.assertEquals(101, removeEvents[0].getData(1));
+                        AssertJUnit.assertEquals(102, removeEvents[1].getData(1));
+                        innerAssertionsPassed = true;
+                    } else if (removeEventCount == 4) {
+                        innerAssertionsPassed = false;
+                        AssertJUnit.assertTrue("user0".equals(removeEvents[0].getData(0)));
+                        AssertJUnit.assertTrue("user0".equals(removeEvents[1].getData(0)));
+
+                        AssertJUnit.assertEquals(103, removeEvents[0].getData(1));
+                        AssertJUnit.assertEquals(104, removeEvents[1].getData(1));
+                        innerAssertionsPassed = true;
+                    }
+                }
+                eventArrived = true;
+            }
+
+        });
+
+        InputHandler inputHandler = siddhiAppRuntime.getInputHandler("purchaseEventStream");
+        siddhiAppRuntime.start();
+
+        inputHandler.send(new Object[]{"user0", 101, 34.4, 5});
+        inputHandler.send(new Object[]{"user0", 102, 24.5, 2});
+        Thread.sleep(1100);
```

Comment:
You will have to increase this sleep a bit

Reply:
done accordingly